### PR TITLE
IceGrid: remove dependency on IcePatch2

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR removes IceGrid's dependency on IcePatch2.

IceGrid was using helper functions from IcePatch2's client library. This PR copies these 5 functions to IceGrid's internal Util.h/cpp.

### CODE REVIEW

File: cpp/src/IceGrid/Util.h [hunk 1]
```diff
@@ -38,6 +40,12 @@ namespace IceGrid
 
     int secondsToInt(const std::chrono::seconds&);
 
+    void createDirectory(const std::string&);
```
Comment (bernardnormier) [inline cpp/src/IceGrid/Util.h:43]: All these functions are a straight copy from IcePatch2.

File: cpp/src/IceStorm/Makefile.mk [hunk 2]
```diff
@@ -43,7 +43,7 @@ icestormadmin_sources           := $(addprefix $(currentdir)/,Admin.cpp \
                                                              IceStormInternal.ice)
 
 icestormdb_targetdir            := $(bindir)
-icestormdb_dependencies         := IcePatch2 IceDB
```
Comment (bernardnormier) [inline cpp/src/IceStorm/Makefile.mk:46]: This was a bogus dependency. icestormdb doesn't use IcePatch2 or any of its helper functions.

**Review by pepone — APPROVED** [review]
It looks good, I assume the Util functions are the same as in IcePatch2?

**Comment by bernardnormier** [comment]
> It looks good, I assume the Util functions are the same as in IcePatch2?

Yes, they are identical.

**Review by externl — APPROVED** [review]
Looks good!

**Pull request merged**